### PR TITLE
Video Component Enhancements + Add Close Button For Background Embed

### PR DESCRIPTION
**Pull request by sghoweri**
Polish Brightcove Video Player styles; add close button functionality when videos embedded as background.

- Hard code adding 'zero' value spacing utility classes (ex. `u-bolt-margin-right-none`) -- currently not being auto-generated as a Utility class like all the other spacing scale values. Workaround till #287 is resolved
- Add Bolt Core CSS Dependencies to @bolt/utilities-spacing to allow the package's Sass to be compiled by itself (or imported into a Component)
- Add new `close` method to the Band component to allow it to be programmatically collapsed via a method call
- Refine animation of Video component when embedded as a background layer (now animating in as a 3D transformed layer)
- Refine expand / collapse animation of the Band component 
- Add close button to Video component to allow it to be closed when embedded inside a Background.
- Update Video component to support a new UIID parameter to internally wire up the new "close video button" to the corresponding video
- Refine the styling of video player embedded as a background -- when a video is "ready", trigger additional style enhancements (ie. allow the background overlay to the video player to be animated separately from the player itself
- Tweak expand / collapse logic of the Band component to account for instances where the embedded content's height / width (ie. the video player) can't be calculated
- Update the Button component to support new `rounded` and `iconOnly` paramaters
- Update the Button Component to render based on the params passed into the custom element
- Hack the Brightcove Player's  default styles to address responsiveness and usability issues with the Social Share overlay. Without this, the social share overlay in Brightcove awkwardly isn't vertically or horizontally centered, doesn't have a max width, and the close button directly interferes with the new "collapse Band" close button being added.

### CODE REVIEW

File: src/_patterns/02-components/bolt-video/src/video.twig [hunk 1]
```diff
@@ -21,21 +21,21 @@
 
 {% set videoTag %}
   {% include "@bolt/_video-tag.twig" %}
-  {% embed "@bolt/wrapper.twig" with {
```
Comment (remydenton) [inline src/_patterns/02-components/bolt-video/src/video.twig]: +1 For eliminating the embed

File: src/_patterns/02-components/bolt-video/src/_videojs-enhancements.scss [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Vertically and horizontally center the default Brightcove share overlay content
```
Comment (remydenton) [inline src/_patterns/02-components/bolt-video/src/_videojs-enhancements.scss]: Good call breaking this out into its own file.

File: src/_patterns/01-core/07-utilities/utilities-spacing/_utilities.spacing.scss [hunk 3]
```diff
@@ -5,6 +5,48 @@ $bolt-export-data: false !global;
...
+}
+
+.u-bolt-padding-top-none {
+  padding-top: 0 !important;
+}
+
+.u-bolt-padding-bottom-none {
+  padding-top: 0 !important;
+}
+
```
Comment (remydenton) [inline src/_patterns/01-core/07-utilities/utilities-spacing/_utilities.spacing.scss]: Sometimes you just have to git 'er done :).

Is there an open ticket to come back and address this?

Comment (sghoweri) [reply src/_patterns/01-core/07-utilities/utilities-spacing/_utilities.spacing.scss]: @remydenton yuuuup. exactly :)

Open ticket already opened on this here: https://github.com/bolt-design-system/bolt/issues/287

**Pull request merged**